src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,34 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.Collections;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<? extends Machine> getAll(Class type) {
+        if (type == Bulldozer.class) {
+            return new BulldozerProducer().get();
+        }
+        if (type == Excavator.class) {
+            return new ExcavatorProducer().get();
+        }
+        if (type == Truck.class) {
+            return new TruckProducer().get();
+        }
+        return Collections.emptyList();
```

Comment:
lets create producers as class-level variables - to avoid creating a new instance of producer each time this method is invoked
ps. remember about OOP principles, use abstract reference type